src/shared/parse.cpp

```diff
@@ -1351,19 +1351,14 @@ Parse_CopyToken
 */
 static token_t *Parse_CopyToken( const token_t *token )
 {
-	token_t *t;
-
-//  t = (token_t *) malloc(sizeof(token_t));
-	t = ( token_t * ) Z_Malloc( sizeof( token_t ) );
+	token_t *t = new (std::nothrow) token_t( *token );
```

Comment:
> By default, when the new operator is used to attempt to allocate memory and the handling function is unable to do so, a [bad_alloc](https://www.cplusplus.com/bad_alloc) exception is thrown. But when nothrow is used as argument for new, it returns a null pointer instead.

This is not what we want. We want the game to immediately crash when it runs out of memory, which an unhandled exception accomplishes. Returning a null pointer means the game crashes at some later point which may make it hard to trace where the failed allocation occurs.

Reply:
Immediate crashes are fine to me, except when the only error message is "Pipe broken", which happens fairly often to me.
I think using `Sys::Drop()` would be better than to rely on exceptions being replaced by "abort()" here.
Ideally, [[nodiscard]] would have been nice (not only for those ones) but it's a C++17 feature, and I guess we can't use it, right?

Reply:
> I think using Sys::Drop() would be better than to rely on exceptions being replaced by "abort()" here.

Actually we already do this via our `operator new` override in daemon/src/common/System.cpp.

> Ideally, [[nodiscard]] would have been nice (not only for those ones) but it's a C++17 feature, and I guess we can't use it, right?

You can't use the spelling `[[nodiscard]]`, but vendor-specific annotations with equivalent functionality are available in earlier versions. If you want that you can add the appropriate defines for each supported compiler in daemon/src/common/Compiler.h.

Reply:
Actually we already have one, `WARN_UNUSED_RESULT`

Reply:
For now I'll just do the crash thing though. I dream of getting rid of that parsing code anyway, considering how unreliable it is, so just fixing the memleak bugs shall be enough, especially since this is the scope of this PR.